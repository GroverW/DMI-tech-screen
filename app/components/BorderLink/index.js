/**
 *
 * BorderLink
 *
 */

import styled from 'styled-components';
import { Link } from 'react-router-dom';

const BorderLink = styled(Link)`
  padding: 7px 10px;
  margin: 0px 10px;
  color: ${props => props.theme.font};
  border: 2px solid ${props => props.theme.buttonBorder};
  border-radius: 5px;
  text-decoration: none;
  &:hover {
    background-color: ${props => props.theme.primaryHover};
    color: white;
  }
  &:focus {
    background-color: ${props => props.theme.primaryFocus};
    color: white;
  }
`;

export default BorderLink;
