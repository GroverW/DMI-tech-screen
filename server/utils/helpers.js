const { RESULTS_PER_PAGE } = require('../config');

const getPageStart = page =>
  page && Number.isInteger(+page)
    ? Math.max(page - 1, 0) * RESULTS_PER_PAGE
    : 0;

module.exports = {
  getPageStart,
};
