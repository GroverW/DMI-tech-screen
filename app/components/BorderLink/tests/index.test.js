/**
 *
 * Tests for BorderLink
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { renderWithRouter } from '../../../tests/helpers/testHelper';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import BorderLink from '../index';

describe('<BorderLink />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderWithRouter(<BorderLink to="" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithRouter(<BorderLink to="" />);
    expect(firstChild).toMatchSnapshot();
  });
});
