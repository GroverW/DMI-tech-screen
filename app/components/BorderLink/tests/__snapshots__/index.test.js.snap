// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BorderLink /> Should render and match the snapshot 1`] = `
<a
  class="BorderLink-sc-1brm82f-0 hASwHg"
  href="/"
/>
`;
